src/eins-hwinfo.c

```diff
@@ -105,9 +115,21 @@ round_to_nearest (guint64 size,
   return (guint32) CLAMP (size / divisor, 0, G_MAXUINT32);
 }
 
-GVariant *
-eins_hwinfo_get_disk_space_for_partition (GFile   *file,
-                                          GError **error)
+guint32
+eins_hwinfo_get_ram_size (void)
+{
+  glibtop_mem mem;
+  guint32 size_mib;
+
+  glibtop_get_mem (&mem);
+  size_mib = round_to_nearest (mem.total, ONE_MIB_IN_BYTES);
+  return size_mib;
+}
+
+void
+eins_hwinfo_get_disk_space_for_partition (GFile          *file,
+                                          DiskSpaceType  *diskspace,
+                                          GError        **error)
 {
```

Comment:
```suggestion
{
g_return_val_if_fail (diskspace != NULL, FALSE);
```

(but moved below the variable declarations)

Reply:
ok.  But, fixed with a little bit different way for "the other option" mentioned above.